subsys/bluetooth/host/id.c

```diff
@@ -1747,9 +1751,12 @@ int bt_id_set_create_conn_own_addr(bool use_filter, uint8_t *own_addr_type)
 			if (err) {
 				return err;
 			}
-		}
 
-		*own_addr_type = addr->type;
+			*own_addr_type = BT_HCI_OWN_ADDR_RANDOM;
+		} else {
+			/* Privacy is not enabled, so if address type is not random, it's public */
```

Comment:
Indeed. 

Why do we even have 
```
#define BT_ADDR_LE_PUBLIC_ID    0x02
#define BT_ADDR_LE_RANDOM_ID    0x03
```
?

They seem completely unused except for `bt_addr_le_from_str`

Reply:
They are not used directly anywhere. I think we can remove them. `ADDR_RESOLVED_BITMASK` is used instead.

However, it looks like we are abusing types:

https://github.com/zephyrproject-rtos/zephyr/blob/03959a20f7834ffe50a51df2b2e45ef87d1b610b/include/zephyr/bluetooth/hci_types.h#L2963

The above is not a valid `bt_addr_le_t`, since it can have that extra bit set. It seems addresses are cleaned up (using `bt_addr_le_copy_resolved`) most of the time before being given to the application. But worryingly, it looks like some code paths lead to this stuff bubbling up to the application.